src/cmake/logging.cmake

```diff
@@ -1,4 +1,5 @@
 
+if(NOT EXISTS ${CMAKE_SOURCE_DIR}/config.h)
```

Comment:
Should this be unconditional in case there are changes since the last run?

Reply:
Ah, well, very good point. The logic here has to prevent any generation/git querying if we are building from sdist, so we don't have `.git` in folder. I have another idea, I can check if there is .git folder and only generate that file in that case.

Reply:
SGTM thanks